blog/extractImports.ts

```diff
@@ -0,0 +1,43 @@
+import path from "path";
+import fs from "fs";
+
+/**
+ *
+ * @param origin - the root of the filesystem where this function is being called. Important
+ * because `fs` will use it to find the dependencies of the blog post
+ * @param blogContents - the blog post raw contents
+ * @returns {Record<string, string>} - the dependencies of the blog in a dictionary matching relative file
+ * path to dependency source content
+ */
+export function extractImports(
```

Comment:
if i understand this one right, this is to gather up any dependent components that are used in the mdx, right?

i'd vote to not worry about finding the comps based on the contents of the mdx. j/ bring in all the comps from `/blog/posts/components`. if a comp isn't used, i don't think it'll be that big of a deal perf wise and in exchange you can drop the regex and path code, thus reducing the chance for bugs.

if you keep this, i think some naming tweaks(`findDependentComps` perhaps to be more precise? `origin` seems like a `rootDir` although i think the `rootDir` is always `POSTS_SOURCE_DIR`) and some heavy testing would be beneficial here.

Reply:
This seems like another situation of, "What kind of solution are we trying to make here, just for us, or possibly for others?" Writing the regex was about making sure we only got the stuff needed for the specific post we care about, so it was made to be super-efficient. For Remix Austin, it probably doesn't matter because the site is so small. But if we want to ship this for others, I wonder if having that performance would help with adoption.

I'm down for removing this. We have a record of it in case we need it again.